### ISSUE TITLE
Wrong error elaboration for missing property in union type involving index signature

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#31306]
**TypeScript Version:**  3.3.3

**Search Terms:**

property does not exist, union, index signature

**Code**

```ts
declare const o: Record<string, string> | {prop: string};
o.foo; // wrong error message

declare const o2: {prop: number} |  Record<string, number>;
o2.foo; // correct error message
```

**Expected behavior:**

`o.foo` should give the following error:

```
Property 'foo' does not exist on type '{ prop: string; } | Record<string, string>'.
  Property 'foo' does not exist on type '{ prop: string; }'.
```

**Actual behavior:**

Depending on check-order (in this example caused by the order of the union constituents) `o.foo` gives this error, which is clearly wrong:

```
Property 'foo' does not exist on type 'Record<string, string> | { prop: string; }'.
  Property 'foo' does not exist on type 'Record<string, string>'.
```

**Playground Link:** https://typescript-play.js.org/#code/CYUwxgNghgTiAEYD2A7AzgF3kgXPASuEjMADyYwCWKA5gDTwXU0B88APvAN4AOMSPPE1oBfANwAoJADoAZkiSSJoSLATJ0WJACY8vfoPgoArgFsARiBgiO8AkRLkMVWgxMWrLSTrkKxQA

**Related Issues:**
